bin/live/pycbc_live_single_trigger_fits

```diff
@@ -198,110 +191,219 @@ for filename in files:
 
-logging.info("All events processed")
+            # Read trigger value datasets from file
+            triggers[ifo] = {k: fin[ifo][k][:] for k in fin[ifo].keys()
+                             if k not in ('loudest', 'stat', 'gates', 'psd')
+                             and fin[ifo][k].size == n_triggers}
+            # The n_triggers filter should be enough, but the 'not in'
+            # list is for edge cases where the number of gates/loudest etc is
+            # the same as the number of triggers
 
+            # The stored chisq is actually reduced chisq, so we need to
+            # alter the chisq_dof dataset so we can use the standard conversions.
+            # chisq_dof of 1.5 gives the right number (2 * 1.5 - 2 = 1)
+            triggers[ifo]['chisq_dof'] = \
+                1.5 * np.ones_like(triggers[ifo]['snr'])
 
-# calculate new_snr and cut if requested
-if cutting_newsnr or args.fit_param == 'new_snr' or 'new_snr' not in events[ifo].data:
-    logging.info("Calculating new_snr")
-    for ifo in args.ifos:
-        events[ifo].data['new_snr'] = ranking.newsnr(events[ifo].data['snr'],
-                                                     events[ifo].data['chisq'])
-#Note: according to Alex, the stored chisq here is already the reduced chisq
 
-if cutting_newsnr:
-    logging.info("Keeping events with new_snr %s %.3f" % (nsineq[0], nsthresh))
-    ineq_fn = inequality_string_to_function(nsineq)
     for ifo in args.ifos:
-        idx_keep = np.nonzero(ineq_fn(events[ifo].data['new_snr'], nsthresh))[0]
-        if len(idx_keep):
-            events[ifo] = events[ifo].select(idx_keep)
+        # Continue if not triggers found in file
+        if ifo not in triggers:
+            continue
+
+        # Apply the cuts to triggers
+        keep_idx = cuts.apply_trigger_cuts(triggers[ifo], trigger_cut_dict)
+
+        # triggers contains the datasets that we want to use for
+        # the template cuts, so it can be used as the template bank
+        # in this use case
+        keep_idx = cuts.apply_template_cuts(triggers[ifo], template_cut_dict,
+                                            template_ids=keep_idx)
+
+        # Skip if no triggers survive the cuts
+        if not keep_idx.size: continue
+
+        # Apply the cuts
+        triggers_cut = {k: triggers[ifo][k][keep_idx]
+                        for k in triggers[ifo].keys()}
+
+        # Calculate the sngl_ranking value to be used in the fits
+        sngls_value = ranking.get_sngls_ranking_from_trigs(
+                          triggers_cut, args.sngl_ranking)
+
+        triggers_cut[args.sngl_ranking] = sngls_value
+
+        triggers_da = DictArray(data=triggers_cut)
+
+        # If we are clustering, take the max sngl_ranking value
+        if args.cluster:
+            # Use sngls_ranking parameter for clustering
+            max_idx = sngls_value.argmax()
+            triggers_cut = triggers_da.select(max_idx)
+
+        if no_triggers_yet[ifo]:
```

Comment:
Do you really need the auxiliary `no_triggers_yet` dict? Can you not do this instead:
```python
if ifo in events:
    events[ifo] += triggers_da
else:
    events[ifo] = triggers_da
```